cartographer/pose_graph/constraint/constraint_utils.h

```diff
@@ -31,6 +31,13 @@ void AddPose3D(Pose3D* pose, ceres::Problem* problem);
 
 void AddImuCalibration(ImuCalibration* pose, ceres::Problem* problem);
 
+#define FIND_NODE_OR_RETURN(node_name, node_id, map, log_message) \
```

Comment:
Since this is defined in a public header, maybe prefix the macro with `CARTOGRAPHER_`?

This is the first time that Cartographer defines a macro for something, though. Are you sure the readability is worth doing this?

Reply:
Later (end of q3), when absl has statusor lib, we will migrate and remove this code.